### PR TITLE
Add Swtich to disable NetworkConfigurationChanges

### PR DESCRIPTION
**Pull request by ZhidongPeng**
## Description

Issue # 

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and Travis.CI is passing.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: tests/ga/test_monitor.py [hunk 1]
```diff
@@ -58,28 +58,34 @@ def periodic_operation_run(self):
         invoked_operations = []
 
         with _mock_wire_protocol():
-            with patch("azurelinuxagent.ga.monitor.MonitorHandler.stopped", side_effect=[False, True]):
+            with patch("azurelinuxagent.ga.monitor.MonitorHandler.stopped", side_effect=[False, True, False, True]):
```
Comment (narrieta) [inline tests/ga/test_monitor.py:61]: I think this should be [ False, False, True ] (i.e. do 2 iterations of the main loop and then stop)

File: tests/ga/test_monitor.py [hunk 2]
```diff
@@ -58,28 +58,34 @@ def periodic_operation_run(self):
...
+                                expected_operations = [
+                                    'poll resource usage',
+                                    'report network errors',
+                                    'reset periodic log messages',
+                                    'send_host_plugin_heartbeat',
+                                    'send_imds_heartbeat'
+                                ]
 
-                        self.assertEqual(invoked_operations, expected_operations, "The monitor thread did not invoke the expected operations")
+                                if monitor_network_changes():
```
Comment (narrieta) [inline tests/ga/test_monitor.py:82]: Invoking the mock here doesn't feel quite right, I'd suggest using the loop variable instead (network_changes)

**Review by narrieta — COMMENTED** [review]
2 small comments in the test code, otherwise LGTM

File: tests/ga/test_monitor.py (hunk 1 above)
Comment (ZhidongPeng) [reply tests/ga/test_monitor.py:61]: yes, we need run it with 2 iterations, while it requires both **run and join** at each iteration. so it has to be True to allow the monitor thread only run once per iteration. if I set to False, False, True, the invoked_operations get doubled and it only runs for the first iteration.

File: tests/ga/test_monitor.py (hunk 2 above)
Comment (ZhidongPeng) [reply tests/ga/test_monitor.py]: make sense, I updated it.

File: tests/ga/test_monitor.py (hunk 1 above)
Comment (narrieta) [reply tests/ga/test_monitor.py:61]: yup, you are correct

**Review by narrieta — APPROVED** [review]

**Pull request merged**